lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -846,6 +846,17 @@ class DxilFinalizeModule : public ModulePass {
         MarkUsedSignatureElements(DM.GetEntryFunction(), DM);
         if (DM.GetShaderModel()->IsHS())
           MarkUsedSignatureElements(DM.GetPatchConstantFunction(), DM);
+      } else {
+        for (auto &function : M.getFunctionList()) {
+          if (!function.isDeclaration()) {
+            if (DM.HasDxilEntryProps(&function)) {
+              const auto &entryProps = DM.GetDxilEntryProps(&function);
+              if (entryProps.props.IsMeshNode()) {
```

Comment:
Is there a reason this needs to be limited to mesh node shaders?

Reply:
Note, if you were to open it up to other shader stages, you would want to use the same pattern for hull shader I suggested.  In fact that little bit could be integrated into the beginning of the MarkUsedSignatureElements function.  Something like:
```cpp
// For HS, also traverse patch constant function
if (entryProps.props.IsHS() && entryProps.props.ShaderProps.HS.patchConstantFunc && entryProps.props.ShaderProps.HS.patchConstantFunc != F)
  MarkUsedSignatureElements(entryProps.props.ShaderProps.HS.patchConstantFunc, entryProps);
```

Reply:
Typically, shaders in libs are never linked in a pipeline in the runtime, which is the reasoning for why we didn't set the usage mask for library shaders as far as I know. So, I think it makes sense to limit this to mesh node shaders only.